From 8c8667edc807c3e42a8739924d5c8006e07e2995 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 15 Jan 2019 16:40:49 +0200
Subject: [PATCH] [SW-0] pll: fix the CPU PLL config condition

The hl_pll_is_bypass() semantics are to return the bypass value and
not the true or false, as opposed to its name which is thus misleading.

Fix the CPU PLL bypass check bug, resulted from the ambiguity of the
hl_pll_is_bypass() function name.
Change the function name from hl_pll_is_bypass() to hl_pll_bypass_get()
thus resolving the ambiguity.

Fixes c33fdb4 ([SW-0] pll: configure CPU PLL only if in bypass).

Change-Id: Idb6cf070418fde84f55e7f041fc76fd639477ea2
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/pll.c b/pll.c
index 1fc705b..55008f7 100644
--- a/pll.c
+++ b/pll.c
@@ -77,7 +77,7 @@
 	return pll_num;
 }
 
-static enum hl_pll_bypass_val hl_pll_is_bypass(struct hl_pll_cfg *cfg, enum hl_pll_div_out out)
+static enum hl_pll_bypass_val hl_pll_bypass_get(struct hl_pll_cfg *cfg, enum hl_pll_div_out out)
 {
 	int offset = out * sizeof(u32);
 
@@ -92,7 +92,7 @@
 	 * 1) the CPU PLL offset in the array is 0
 	 * 2) DIV1 is the PLL output to timestamp counter.
 	 */
-	enum hl_pll_bypass_val bval = hl_pll_is_bypass(goya_pll_cfg, HL_PLL_DIV1);
+	enum hl_pll_bypass_val bval = hl_pll_bypass_get(goya_pll_cfg, HL_PLL_DIV1);
 
 	if (bval == HL_PLL_REF_CLK || bval == HL_PLL_DIV_REF)
 		cnt = BYPASS_CYCLES_PER_USEC;
@@ -228,12 +228,15 @@
 {
 	int i, pll_num;
 	struct hl_pll_cfg *hl_pll_cfg;
+	enum hl_pll_bypass_val bval;
 
 	pll_num = hl_pll_get_cfg(&hl_pll_cfg);
 
 	for (i = 0; i < pll_num; i++) {
+		/* DIV0 is the undivided PLL output */
+		bval = hl_pll_bypass_get(hl_pll_cfg + i, HL_PLL_DIV0);
 		if ((u64)hl_pll_cfg[i].base == mmCPU_PLL_BASE &&
-			!hl_pll_is_bypass(hl_pll_cfg + i, HL_PLL_DIV0))
+			bval != HL_PLL_REF_CLK && bval != HL_PLL_DIV_REF)
 				continue;
 
 		hl_pll_set(hl_pll_cfg + i);
